src/Microsoft.AspNetCore.Mvc.Razor.Host/Internal/ViewComponentTagHelperChunkVisitor.cs

```diff
@@ -0,0 +1,228 @@
+                foreach (var descriptor in viewComponentDescriptors)
+                {
+                    base.Accept(chunk);
+                    return;
+                }
+            }
+        }
+
+        /// <summary>
+        /// We generate classes for each view component <see cref="TagHelperDescriptor"/> in the chunk.
+        /// </summary>
+        /// <param name="chunk">The <see cref="TagHelperChunk"/> to visit.</param>
+        protected override void Visit(TagHelperChunk chunk)
+        {
+            var viewComponentDescriptors = chunk.Descriptors.Where(
+                descriptor => ViewComponentTagHelperDescriptorConventions.IsViewComponentDescriptor(descriptor));
+
+            foreach (var descriptor in viewComponentDescriptors)
+            {
+                var descriptorName =
+                    descriptor.PropertyBag[ViewComponentTagHelperDescriptorConventions.ViewComponentNameKey];
+                if (!_writtenViewComponents.Contains(descriptorName))
+                {
+                    _writtenViewComponents.Add(descriptorName);
+                    WriteClass(descriptor);
+                }
+            }
+        }
+
+        private void WriteClass(TagHelperDescriptor descriptor)
+        {
+            // Add target element.
+            BuildTargetElementString(descriptor);
+
+            // Initialize declaration.
+            var tagHelperTypeName = $"{_context.TagHelpersNamespace}.{nameof(TagHelper)}";
+
+            var shortName = descriptor.PropertyBag[ViewComponentTagHelperDescriptorConventions.ViewComponentNameKey];
+            var className = $"__Generated__{shortName}ViewComponentTagHelper";
+
+            using (Writer.BuildClassDeclaration("public", className, new[] { tagHelperTypeName }))
+            {
+                // Add view component helper for reasons.
+                Writer.WriteVariableDeclaration(
+                    $"private readonly {_context.IViewComponentHelperType}",
+                    $"_viewComponentHelper", "");
+
+                // Add constructor.
+                BuildConstructorString(className);
+
+                // Add attributes.
+                BuildAttributeDeclarations(descriptor);
+
+                // Add process method.
+                BuildProcessMethodString(descriptor);
+            }
+        }
+
+        private void BuildConstructorString(string className)
+        {
+            KeyValuePair<string, string> helperPair = new KeyValuePair<string, string>(
+                _context.IViewComponentHelperType,
+                "viewComponentHelper");
+
+            using (Writer.BuildConstructor(
+                "public",
+                className,
+                new List<KeyValuePair<string, string>>() { helperPair }))
+            {
+                Writer.WriteVariableDeclaration("", "_viewComponentHelper", "viewComponentHelper");
```

Comment:
Is this correct? Looks like it'd declare a variable instead of assign a variable.